includes/Modules/Analytics/Advanced_Tracking/Measurement_Event_Factory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Event_Factory
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking;
+
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Measurement_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Woocommerce_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\WPForms_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\CF7_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\FormidableForms_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\NinjaForms_Event_List;
+
+/**
+ * Class for producing Measurement_Event objects based on currently active plugins.
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Measurement_Event_Factory {
+
+	/**
+	 * Instantiates a subclass of MeasurementEventList based on the given plugin name.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param string $plugin_name represents the plugin name to create a list of events for.
+	 * @return Measurement_Event_List
+	 */
+	public function create_measurement_event_list( $plugin_name ) {
+		$event_list = null;
+		switch ( $plugin_name ) {
+			case 'WooCommerce':
+				$event_list = new Woocommerce_Event_List();
+				break;
+			case 'WPForms Lite':
+			case 'WPForms':
+				$event_list = new WPForms_Event_List();
+				break;
+			case 'Contact Form 7':
+				$event_list = new CF7_Event_List();
+				break;
+			case 'Formidable Forms':
+				$event_list = new FormidableForms_Event_List();
+				break;
+			case 'Ninja Forms':
+				$event_list = new NinjaForms_Event_List();
+				break;
+		}
+		return $event_list;
+	}
+
+}
```

Comment:
Instead of having this class and `switch` clause, we could simplify the code path by providing the class name for a plugin's `*_Event_List` within the `$supported_plugins` configuration. This entire class wouldn't be necessary then, and also it would keep the plugins definition coupled to having a class for it, which at the moment isn't immediately obvious.